server_test.go

```diff
@@ -66,11 +66,10 @@ func RunLocalUDPServerWithFinChan(laddr string) (*Server, string, chan struct{},
 	waitLock.Lock()
 	server.NotifyStartedFunc = waitLock.Unlock
 
-	fin := make(chan struct{}, 0)
+	fin := make(chan error, 1)
```

Comment:
why did you make this buffered?

Reply:
@miekg It needs to be buffered in case fin is never read from. (This can happen if Shutdown fails or the test cases timeout). If it was left unbuffered, the goroutine below would block forever.

Reply:
Can we add a comment of this nature?

Reply:
Will do.

Reply:
@miekg I spent far longer than I care to admit figuring out what to say. Hopefully b0869bd is good.